rllib/env/single_agent_episode.py

```diff
@@ -230,15 +230,15 @@ def __init__(
         infos = infos or [{} for _ in range(len(observations or []))]
 
         # Observations: t0 (initial obs) to T.
-        self.observation_space = observation_space
+        self._observation_space = None
```

Comment:
We need to add a `module_id` and `agent_id`. Like this we could treat everything as MA.

Reply:
This will be part of a follow-up PR, in which we make use of the new single-agent/multi-agent unification methods for ConnectorV2, e.g. `ConnectorV2.single_agent_episodes_iterator`, which allows users to build connectors easily that cover both multi-agent AND single-agent cases in one piece of code (no if-else blocks required).